src/ngraph/runtime/cpu/builder/random_uniform.cpp

```diff
@@ -0,0 +1,160 @@
+                auto arg1_buffer_index =
+                    external_function->get_buffer_index(args[1].get_name()); // max_val
+                auto arg2_buffer_index =
+                    external_function->get_buffer_index(args[2].get_name()); // output_shape
+                auto arg3_buffer_index =
+                    external_function->get_buffer_index(args[3].get_name()); // use_fixed_seed
+
+                auto out_buffer_index = external_function->get_buffer_index(out[0].get_name());
+                size_t element_count = out[0].get_size();
+
+                auto index = external_function->add_state(new ngraph::UniformRNGState());
+                auto fixed_seed = ru->get_fixed_seed();
+
+                functor = [&,
+                           index,
+                           element_count,
+                           arg0_buffer_index,
+                           arg1_buffer_index,
+                           arg2_buffer_index,
+                           arg3_buffer_index,
+                           out_buffer_index,
+                           fixed_seed](CPURuntimeContext* ctx, CPUExecutionContext* /* ectx */) {
+                    // TODO: get shape when required
+
+                    T min_val = static_cast<T*>(ctx->buffer_data[arg0_buffer_index])[0];
+                    T max_val = static_cast<T*>(ctx->buffer_data[arg1_buffer_index])[0];
+                    bool use_fixed_seed = static_cast<bool>(
+                        static_cast<char*>(ctx->buffer_data[arg3_buffer_index])[0]);
+
+                    if (!use_fixed_seed)
+                    {
+                        reference::random_uniform<T>(
+                            static_cast<T*>(ctx->buffer_data[out_buffer_index]),
+                            min_val,
+                            max_val,
+                            element_count,
+                            static_cast<UniformRNGState*>(ctx->states[index]));
+                    }
+                    else
+                    {
+                        reference::random_uniform_with_fixed_seed<T>(
+                            static_cast<T*>(ctx->buffer_data[out_buffer_index]),
+                            min_val,
+                            max_val,
+                            element_count,
+                            fixed_seed);
+                    }
+                };
+                return functor;
+            }
+
+            template <>
+            void Builder::BUILDER_DECL(ngraph::op::RandomUniform)
+            {
+                auto& functors = external_function->get_functors();
+                CPUKernelFunctor functor;
+                if (args[2].get_element_type() != element::i64)
+                {
+                    throw ngraph_error("Unsupported index 2 element type");
+                }
+                auto element_type = args[0].get_element_type();
+                if (element_type == element::f32)
```

Comment:
Use a switch statement and set the #pragma flags so that all element types are required to be here or it won't compile. That way you don't get a runtime error for not supported element type